lua/rustaceanvim/lsp/init.lua

```diff
@@ -249,11 +250,55 @@ M.reload_settings = function(bufnr)
   return clients
 end
 
+---Updates the target architecture setting for the LSP client associated with the given buffer.
+---@param bufnr? number The buffer number, defaults to the current buffer
+---@param target? string The target architecture. Defaults to the current buffer's target if not provided.
+M.set_target_arch = function(bufnr, target)
+  local function update_target(client)
+    -- Get the current target from the client's settings
+    local current_target = client.config.settings
+      and client.config.settings['rust-analyzer']
+      and client.config.settings['rust-analyzer'].cargo
+      and client.config.settings['rust-analyzer'].cargo.target
+
+    if not target then
+      if not current_target then
+        vim.notify('Using default OS target architecture.', vim.log.levels.INFO)
+      else
+        vim.notify('Target architecture is already set to the default OS target.', vim.log.levels.INFO)
+      end
+      return
+    end
+
+    if targets.target_is_valid_rustc_target(target) then
+      client.settings['rust-analyzer'].cargo.target = target
+      client.notify('workspace/didChangeConfiguration', { settings = client.config.settings })
+      vim.notify('Target architecture updated successfully to: ' .. target, vim.log.levels.INFO)
+      return
+    else
+      -- TODO: Log error when logging is implemented
+      vim.notify('Invalid target architecture provided: ' .. tostring(target), vim.log.levels.ERROR)
+      return
+    end
+  end
+
+  M.restart_impl(bufnr, update_target)
+end
+
 ---Restart the LSP client.
 ---Fails silently if the buffer's filetype is not one of the filetypes specified in the config.
 ---@param bufnr? number The buffer number (optional), defaults to the current buffer
 ---@return number|nil client_id The LSP client ID after restart
 M.restart = function(bufnr)
+  M.restart_impl(bufnr)
+end
+
+---LSP restart ininer implementations
+---@param bufnr? number
+---@param set_target_callback? function(client) Optional callback to run for each client before restarting.
+---@return number|nil client_id
+M.restart_impl = function(bufnr, set_target_callback)
+  -- TODO: handle case for when requested target change is the same as client active active so we don't restart clients unnecessarily
```

Comment:
issue: I'm no longer a fan of TODO comments, as they often never get done but don't get removed from the code base.
An issue would be better if you don't want to implement it here.

Reply:
you can assign #543 to me